cratedb_toolkit/model.py

```diff
@@ -120,3 +123,44 @@ class InputOutputResource:
     url: str
     format: t.Optional[str] = None  # noqa: A003
     compression: t.Optional[str] = None
+
+
+@define
+class AddressPair:
+    """
+    Manage two URL instances, specifically a pair of source/target URLs,
+    where target is mostly a CrateDB Server, while source is any.
+    """
+
+    source_url: URL
+    target_url: URL
+
+    _source_url_query_parameters: t.Dict[str, t.Any] = Factory(dict)
+    _target_url_query_parameters: t.Dict[str, t.Any] = Factory(dict)
+
+    __SERVER_SCHEMES__ = ["http", "https", "mongodb", "mongodb+srv"]
+
+    def navigate(self, source_path: str, target_path: str) -> "AddressPair":
+        source_url_query_parameters = self.source_url.query_params
+        target_url_query_parameters = self.target_url.query_params
+
+        source_url = URL(str(self.source_url))
```

Comment:
I guess there is no better API at the URL class to avoid string generation + parsing only to copy the instance, right?

Reply:
It looks like standard `deepcopy` is just fine, as already applied on other spots in this file. Improved with 2a5df04467b.